doc/external-api/README.md

```diff
@@ -40,24 +40,37 @@ Nix expression `builtins.nixVersion`.
 #include <nix_api_expr.h>
 #include <nix_api_value.h>
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 
 // NOTE: This example lacks all error handling. Production code must check for
 // errors, as some return values will be undefined.
-int main() {
-   nix_libexpr_init(NULL);
 
-   Store* store = nix_store_open(NULL, "dummy://", NULL);
-   EvalState* state = nix_state_create(NULL, NULL, store); // empty search path (NIX_PATH)
-   Value *value = nix_alloc_value(NULL, state);
+void my_get_string_cb(const char * start, unsigned int n, char ** user_data)
+{
+    *user_data = strdup(start);
+}
```

Comment:
We could provide this function as part of the C interface. It's an obvious choice for C code, and I think most FFIs have something for dealing with malloced memory.